components/sup/src/manager/service/mod.rs

```diff
@@ -447,82 +407,44 @@ impl Service {
                 None
             }
         };
-        let mut rumor = ServiceRumor::new(self.local_member_id.clone(),
-                                          &self.package().ident,
+        let mut rumor = ServiceRumor::new(self.sys.member_id.as_str(),
+                                          &self.pkg.ident,
                                           &self.service_group,
-                                          &*self.config.sys,
+                                          &self.sys.as_sys_info(),
                                           exported.as_ref());
         rumor.set_incarnation(incarnation);
         rumor
     }
 
     /// Run initialization hook if present
-    fn initialize(&mut self) {
+    fn initialize(&mut self, census_ring: &CensusRing) {
         if self.initialized {
             return;
         }
         outputln!(preamble self.service_group, "Initializing");
-        self.hooks.compile(&self.service_group, &self.config);
-        if let Some(err) = self.copy_run().err() {
-            outputln!(preamble self.service_group, "Failed to copy run hook: {}", err);
+        {
+            let ctx = self.render_context(&census_ring);
```

Comment:
Yeah you're right, this actually can completely be avoided. It's already being called in the `update_templates` function that I've created from refactored bits of `update_configuration`. Gonna make those changes and push them up!